### ISSUE TITLE
Column name equals to package name prefix

### ISSUE THREAD

**Issue by elsennov** [Raizlabs/DBFlow#956]
DBFlow Version: 3.1.1
Issue Kind Bug
Description:

Hi, i've an error if one of my column equals to prefix of my package. My package is `id.elsen.mylibrary`, while in my table there is a column named `id`. It will make my application failed to be compiled, it says `cannot find symbol variable elsen`

After i looked into the generated table class it was generated like this

```
/**
 * This is generated code. Please do not modify */
public final class ChatDbModel_Table {
  public static final PropertyConverter PROPERTY_CONVERTER = new PropertyConverter(){ 
  public IProperty fromName(String columnName) {
  return id.elsen.mylibrary.db.ChatDbModel_Table.getProperty(columnName); 
  }
  };

  public static final Property<String> id = new Property<String>(ChatDbModel.class, "id");

  public static final Property<String> conversationId = new Property<String>(ChatDbModel.class, "conversationId");

  public static final IProperty[] getAllColumnProperties() {
    return new IProperty[]{id,conversationId};
  }

  public static BaseProperty getProperty(String columnName) {
    columnName = QueryBuilder.quoteIfNeeded(columnName);
    switch (columnName)  {
      case "`id`":  {
        return id;
      }
      case "`conversationId`":  {
        return conversationId;
      }
      default:  {
        throw new IllegalArgumentException("Invalid column name passed. Ensure you are calling the correct table's column");
      }
    }
  }
}
```

So the `id` will refer to the property variable instead of the package name. Is it already known issue? Thanks

**Comment by trevjonez** [comment]
Seem like an unfortunate collision. I would strongly recommend you change your package name. The concept and term "id" will appear in many more places than just this one model class, as a result you are likely to be frequently qualifying or renaming your variables to avoid the issue.

**Comment by elsennov** [comment]
I am just wondering why does dbflow need to write a complete package at the return section. Why not just `return ChatDbModel_Table` instead of `return id.elsen.mylibrary.db.ChatDbModel_Table`?

**Comment by trevjonez** [comment]
If we look at [TableDefinition.java line 461](https://github.com/Raizlabs/DBFlow/blob/a5429608e77e9895ce0373c15e077bd1560dda70/dbflow-processor/src/main/java/com/raizlabs/android/dbflow/processor/definition/TableDefinition.java#L461)

We should be able to remove the qualifier completely due to the method being called residing in the same class.

After closer inspection that code should also be updated to use the anonymous class builder which provides better formatting for free when compared to the code block that it currently uses. There are probably some other positive gains going with the TypeSpec over the code block also.

Warning I didn't test the bellow code change but it is valid java at least :)
```java
FieldSpec.Builder propertyConverter = FieldSpec.builder(ClassNames.PROPERTY_CONVERTER, "PROPERTY_CONVERTER", Modifier.FINAL, Modifier.PUBLIC, Modifier.STATIC)
            .initializer("$L", TypeSpec.anonymousClassBuilder("")
                .addSuperinterface(ClassNames.PROPERTY_CONVERTER)
                .addMethod(MethodSpec.methodBuilder("fromName")
                    .addAnnotation(Override.class)
                    .addModifiers(Modifier.PUBLIC)
                    .returns(ClassNames.IPROPERTY)
                    .addParameter(String.class, "columnName")
                    .addStatement("return getProperty(columnName)")
                    .build())
                .build());
```

You can make the change and test it by using jitpack to build your fork.
`com.github.elsennov.DBFlow:dbflow-processor:{commit-hash}`

If all is well I would imagine the repo maintainer would accept a pull request with this change.

**Comment by elsennov** [comment]
Ah yes, i will check it and test it. If possible, i will create a PR :)
Thanks @trevjonez !

**Comment by agrosner** [comment]
this is fixed in 4.0.0-beta1/`develop`. I combined the `_Adapter` + `_Table` classes and dont even have that method anymore.

**Comment by agrosner** [comment]
note that 4.0.0-beta1 is not complete and not released yet. Some changes might break your app, and until I make an upgrade guide, I suggest proceeding with caution as you may or may not be affected by breaking changes.
